### PR TITLE
Feat: Project Euler Problem 6

### PR DESCRIPTION
**Pull request by domix80**
Add solution for Euler Problem 6 and change applied also in the directory file to reflect the new problem

### CODE REVIEW

**Comment by domix80** [comment]
@aparibocci, may you check if you can review it?

File: DIRECTORY.md [hunk 1]
```diff
@@ -153,6 +153,8 @@
     * [Sol2](https://github.com/TheAlgorithms/Ruby/blob/master/project_euler/problem_004/sol2.rb)
   * Problem 005
     * [Sol1](https://github.com/TheAlgorithms/Ruby/blob/master/project_euler/problem_005/sol1.rb)
+  * Problem 006
+    * [Sol1](https://github.com/TheAlgorithms/Ruby/blob/master/project_euler/problem_006/sol1.rb)
```
Comment (aparibocci) [inline DIRECTORY.md:157]: DIRECTORY.md is not to be updated, as it gets auto-updated automatically (see https://github.com/TheAlgorithms/Ruby/blob/master/CONTRIBUTING.md)

Comment (aparibocci) [reply DIRECTORY.md:157]:
```suggestion
```

File: project_euler/problem_006/sol1.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#The sum of the squares of the first ten natural numbers #is,
+#    1^2 + 2^2 + ... + 10^2 = 385
+#The square of the sum of the first ten natural numbers #is,
+#    (1 + 2 + ... + 10)^2 = 55^2 = 3025
+#Hence the difference between the sum of the squares of #the first ten
+#natural numbers and the square of the sum is 3025 - 385 = 2640.
+#Find the difference between the sum of the squares of the first one
+#hundred natural numbers and the square of the sum.
+
+def solution?(num)
```
Comment (aparibocci) [inline project_euler/problem_006/sol1.rb:14]: It looks like project euler scripts follow specific guidelines.
Could you please double-check those? https://github.com/TheAlgorithms/Ruby/blob/master/project_euler/README.md

Thank you

**Review by aparibocci — CHANGES_REQUESTED** [review]

**Comment by domix80** [comment]
@aparibocci, Could you please review it?

File: project_euler/problem_006/sol1.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  		y += 1
+  		result += x
+	end
+	r_n_pow2_plus_n_pow2 = result
+	r_sum_n_pow2 = (((num / 2) + 0.5) * num) ** 2
+
+	r_sum_n_pow2 - r_n_pow2_plus_n_pow2
+end
+
+answer = solution?()
```
Comment (aparibocci) [inline project_euler/problem_006/sol1.rb:32]: To comply with https://github.com/TheAlgorithms/Ruby/blob/master/project_euler/README.md

> It should contain positional argument(s) whose default value is the question input. Example: Please take a look at [Problem 1](https://projecteuler.net/problem=1) where the question is to Find the sum of all the multiples of 3 or 5 below 1000. In this case the main solution function will be solution(limit = 1000).

```suggestion
def solution(num=10)
	x = 1
	y = 1
	result = 1
	gap = 3
	while y < num
  		x += gap
  		gap += 2
  		y += 1
  		result += x
	end
	r_n_pow2_plus_n_pow2 = result
	r_sum_n_pow2 = (((num / 2) + 0.5) * num) ** 2

	r_sum_n_pow2 - r_n_pow2_plus_n_pow2
end

answer = solution()
```

**Review by aparibocci — CHANGES_REQUESTED** [review]

File: project_euler/problem_006/sol1.rb (hunk 3 above)
Comment (domix80) [reply project_euler/problem_006/sol1.rb:32]: Ok. got it.

**Review by aparibocci — APPROVED** [review]
Thank you for your contribution

**Comment by domix80** [comment]
@StepfenShawn , May I ask you to review and eventually merge the code?

**Review by StepfenShawn — APPROVED** [review]
Ok, thanks.

**Pull request merged**